zenoh/tests/qos.rs

```diff
@@ -0,0 +1,45 @@
+use std::time::Duration;
+
+use async_std::task;
+use zenoh::{publication::Priority, SessionDeclarations};
+use zenoh_core::{zasync_executor_init, AsyncResolve};
+use zenoh_protocol::core::CongestionControl;
+
+const SLEEP: Duration = Duration::from_secs(1);
+
+#[test]
+fn pubsub() {
+    task::block_on(async {
+        zasync_executor_init!();
+        let session1 = zenoh::open(zenoh_config::peer()).res_async().await.unwrap();
+        let session2 = zenoh::open(zenoh_config::peer()).res_async().await.unwrap();
+
+        let publisher = session1
+            .declare_publisher("test/qos")
+            .res()
+            .await
+            .unwrap()
+            .priority(Priority::DataHigh)
```

Comment:
I find this very weird. 
`Priority` and `CongestionControl` should be set before calling `.res()` and not after.
I wonder if we should really allow that...

Reply:
fixed